buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ErrorReportingTestListener.groovy

```diff
@@ -0,0 +1,121 @@
+package org.elasticsearch.gradle.test
```

Comment:
Sorry, no more new Groovy classes.  
Please convert it to Java, our goal is to get rid of Groovy in buildsrc.

Reply:
FWIW, this class is marked with `@CompileStatic` which means we get all the same type safety and performance benefits of Java, with the conveniences of Groovy. But in general, I think I would prefer and all-Java codebase as well.

I had actually initially started to do this for all classes I so much as modified during this exercise but that quickly got out of hand as code in `src/java` cannot refer to code in `src/groovy` so you eventually hit a point where due to code dependencies, you have to just convert everything. One thing we could do to make this transition easier, and not have to basically move everything in one big to is to actually put _all_ code under `src/groovy` and leverage Groovy [joint compilation](http://groovy-lang.org/groovyc.html#section-jointcompilation). This would enable Groovy to call Java and vice versa. The downside is a performance hit during compilation, but would make it easier to convert things a bit more piecemeal.

In any case, the class in question has no such dependencies, so was trivial to move 😉

Reply:
I'we resisted moving everything to `src/groovy` as being forced to think about the coupling between classes and converting them in this manner leads to a better structure and encourages removing coupling where not necessary (e.x. gluing things together in the build script rather than them knowing about each-other in buildSrc where it makes sense )